File: Packages/Flip54Core/Sources/Flip54Core/Difficulty.swift
public enum Difficulty: String, Codable, CaseIterable, Hashable, Sendable {
    case beginner, standard, advanced

    public var multiplier: Double {
        switch self {
        case .beginner: return 0.5
        case .standard: return 1.0
        case .advanced: return 2.0
        }
    }

    public var displayName: String {
        switch self {
        case .beginner: return "Beginner"
        case .standard: return "Standard"
        case .advanced: return "Advanced"
        }
    }
}

File: android/core/src/main/kotlin/com/flip54/core/Difficulty.kt
package com.flip54.core

import kotlinx.serialization.Serializable

@Serializable
enum class Difficulty {
    BEGINNER, STANDARD, ADVANCED;

    val multiplier: Double get() = when (this) {
        BEGINNER -> 0.5
        STANDARD -> 1.0
        ADVANCED -> 2.0
    }

    val displayName: String get() = when (this) {
        BEGINNER -> "Beginner"
        STANDARD -> "Standard"
        ADVANCED -> "Advanced"
    }
}
